src/PythonProjectBootstrapper/EntryPoint.py

```diff
@@ -177,21 +191,345 @@ def _ExecuteOutputDir(
 
             execute_func = getattr(module, "Execute", None)
             if execute_func:
-                if execute_func(project_dir, output_dir, yes=yes) is False:
+                if execute_func(project_dir, tmp_dir, yes=yes) is False:
                     return
 
+    # generate project in temporary directory so we can avoid overwriting files without user approval
     cookiecutter(
         str(project_dir),
-        output_dir=str(output_dir),
+        output_dir=str(tmp_dir),
         config_file=str(configuration_filename) if configuration_filename is not None else None,
         replay=replay,
         overwrite_if_exists=True,
         accept_hooks=True,
     )
 
+    _CopyToOutputDir(tmp_dir=tmp_dir, output_dir=output_dir)
+    _DisplayPrompts(output_dir=output_dir)
+
 
 # ----------------------------------------------------------------------
+def _CreateManifest(generated_dir: Path) -> dict[Path, str]:
+
+    manifest_dict: dict[Path, str] = {}
+    generated_files: list[Path] = []
+    directories: list[Path] = [generated_dir]
+
+    while directories != []:
+        dir: Path = directories.pop(0)
+        PathEx.EnsureDir(dir)
+
+        for root, dirs, files in os.walk(dir):
+            create_full_path_fn = lambda rel_path: Path(os.path.join(root, rel_path))
+            generated_files += [create_full_path_fn(file) for file in files]
+            directories += [create_full_path_fn(directory) for directory in dirs]
+
+    # Populate manifest dictionary
+    for genfile in generated_files:
+        PathEx.EnsureFile(genfile)
+
+        with open(genfile, "rb") as f:
+            rel_path = PathEx.CreateRelativePath(generated_dir, genfile)
+            manifest_dict[rel_path] = hashlib.file_digest(f, hashlib.sha256).hexdigest()
+
+    return manifest_dict
+
+
 # ----------------------------------------------------------------------
+# Removes any template files no longer being generated as long as the file was never modified by the user
+
+
+def _ConditionallyRemoveUnchangedTemplateFiles(
+    new_manifest_dict: dict[Path, str], existing_manifest_dict: dict[Path, str], output_dir: Path
+) -> None:
+    # files no longer in template
+    removed_template_files: set[Path] = set(existing_manifest_dict.keys()) - set(
+        new_manifest_dict.keys()
+    )
+
+    PathEx.EnsureDir(output_dir)
+
+    # remove files no longer in template if they are unchanged
+    for removed_file_rel_path in removed_template_files:
+        removed_full_path = output_dir / removed_file_rel_path
+
+        if removed_full_path.is_file():
+            with open(removed_full_path, "rb") as removed_file:
+                current_hash = hashlib.file_digest(removed_file, hashlib.sha256).hexdigest()

```

Comment:
I'd suggest making the hash calculation a function that is called from different places in case we want to change the algorithm in the future (for example, we might want to use `sha512` rather than `sha256`).

Reply:
done